pybpodapi/bpod/bpod_base.py

```diff
@@ -271,14 +272,19 @@ def run_state_machine(self, sma):
 
         self.trial_timestamps = []  # Store the trial timestamps in case bpod is using live_timestamps
 
-        self._bpodcom_run_state_machine()
-        if self._new_sma_sent:
+        if self._emulator:
+            self._emulator.set_state_machine(sma)
+            self._emulator.initialize()
+            self._emulator.mirror_state(sma.current_state)
+            # TODO: Do the BpodSystem.RefreshGUI equivalent
+            self.trial_start_timestamp = self._emulator.matrix_start_time
+        else:
+            self._bpodcom_run_state_machine()
             if self._bpodcom_state_machine_installation_status():
                 self._new_sma_sent = False
             else:
                 raise BpodErrorException('Error: The last state machine sent was not acknowledged by the Bpod device.', self)
```

Comment:
One last comment, we've removed line [275](https://github.com/pybpod/pybpod-api/pull/5/files#diff-b440852c8707be42a62338842bdcbe7abac80d379cc70e73fb0e3b3bc02ccc9cL275) above but it wasn't added here. These lines should be 
```suggestion
            if self._new_sma_sent:
                if self._bpodcom_state_machine_installation_status():
                    self._new_sma_sent = False
                else:
                    raise BpodErrorException('Error: The last state machine sent was not acknowledged by the Bpod device.', self)
```